src/client/components/equalizer/equalizer.js

```diff
@@ -76,140 +55,138 @@ class Equalizer extends Component {
+    sound.connect(analyser);
+    const { createAudioData: createAudioDataAsProp } = this.props;
+    createAudioDataAsProp({
+      sound,
+      trackName: file.name,
+      trackType: file.type,
+      trackSize: file.size,
+    });
   }
 
-  startMuteStream = () => {
+   playSoundFromFile = async () => {
+     const { audioData } = this.props;
+     const {
+       sound,
+       playPauseState,
+     } = audioData;
+     const {
+       playPauseSoundFromFile: playPauseSoundFromFileAsProp,
+     } = this.props;
+     if (!playPauseState) {
+       sound.play();
+     } else {
+       sound.pause();
+     }
+     await playPauseSoundFromFileAsProp();
+     await this.renderEqualizer();
+   }
+
+  startMuteStream = async () => {
     const { audioData } = this.props;
     const {
-      audioStream,
-      audioContext,
       analyser,
-      streamSource,
+      voice,
       startMuteState,
     } = audioData;
-
     if (!startMuteState) {
-      streamSource.connect(analyser);
-      analyser.connect(audioContext.destination);
-      // play/pause function doesn't work
-      audioStream.play();
-      this.equaliserRun();
+      voice.connect(analyser);
+      voice.play();
     } else {
-      streamSource.disconnect(analyser);
-      audioStream.pause();
+      voice.pause();
     }
     const { startMuteStreamAudio: startMuteStreamAudioAsProp } = this.props;
-    startMuteStreamAudioAsProp();
-  }
-
-  equaliserRun = () => {
-    requestAnimationFrame(this.renderEqualizer);
+    await startMuteStreamAudioAsProp();
+    await this.renderEqualizer();
   }
 
   renderEqualizer = () => {
     const {
-      analyser, heightArray, ctx, numPoints,
+      analyser, uint8Array, ctx, numPoints,
     } = this.state;
+    const { audioData } = this.props;
+    const { playPauseState, startMuteState } = audioData;
     let isFirstColorForEqualizerUsed = true;
-    analyser.getByteFrequencyData(heightArray);
+    analyser.getByteFrequencyData(uint8Array);
     const { width, height } = ctx.canvas;
     const numbersOfRectengle = 52;
-    const totalAreaOfRectangles = 5 / 6;
+    const totalAreaOfRectangles = 5 / 6 * width;
     const rectangleCornerRadius = 2;
-    const rectangleMaxWidth = 512;
-    const countColumns = Math.floor(width / numbersOfRectengle);
-    const columnWidth = Math.floor(totalAreaOfRectangles * width / numbersOfRectengle);
+    const rectangleMaxHeight = 512;
+    const widthColumnWithPadding = width / numbersOfRectengle;
+    const columnWidth = totalAreaOfRectangles / numbersOfRectengle;
+    const paddingColumn = (widthColumnWithPadding - columnWidth) / 2;
     ctx.clearRect(0, 0, width, height);
-    for (let x = 0; x < width; x += countColumns) {
+    for (let x = 0; x < width - widthColumnWithPadding; x += columnWidth + 2 * paddingColumn) {
       const ndx = Math.floor(x * numPoints / width);
-      const vol = heightArray[ndx];
-      const y = vol * height / rectangleMaxWidth;
+      const vol = uint8Array[ndx];
```

Comment:
No need to shorten your variable names, it’s harder to understand later without context

Reply:
renamed